netty/src/main/java/io/grpc/netty/GrpcSslContexts.java

```diff
@@ -176,5 +187,15 @@ static void ensureAlpnAndH2Enabled(ApplicationProtocolNegotiator alpnNegotiator)
       throw new IllegalArgumentException("This ALPN config does not support HTTP/2. Expected '"
           + HTTP2_VERSION + "', but got " + alpnNegotiator.protocols() + '.');
     }
+    if (!alpnNegotiator.protocols().contains(GRPC_EXP_VERSION)) {
+      throw new IllegalArgumentException("This ALPN config does not support gRPC upgrade. Expected '"
+          + GRPC_EXP_VERSION + "', but got " + alpnNegotiator.protocols() + '.');
+    }
+    if (alpnNegotiator.protocols().indexOf(GRPC_EXP_VERSION)
```

Comment:
Can you explain this?  First protocol should take preference right? Also, if the server prefers H2, thats a valid use case.